pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -215,32 +215,29 @@ func (s *Service) handle(msg *message) error {
 		// TODO pass invitation id #397
 		s.sendMsgEvents(&service.StateMsg{
 			Type: service.PostState, Msg: msg.Msg, StateID: next.Name(),
-			Properties: s.createEventProperties(msg.ThreadID, "")})
+			Properties: createEventProperties(msg.ThreadID, "")})
 		logger.Infof("sent post event for state %s", next.Name())
 
 		next = followup
 	}
 	return nil
 }
 
-func (s *Service) createEventProperties(connectionID, invitationID string) *didExchangeEvent { //nolint: unparam
-	return &didExchangeEvent{connectionID: connectionID, invitationID: invitationID}
-}
-
-// didExchangeEvent implements didexchange.Event interface.
-type didExchangeEvent struct {
-	connectionID string
-	invitationID string
-}
-
-// ConnectionID returns DIDExchange connectionID.
-func (ex *didExchangeEvent) ConnectionID() string {
-	return ex.connectionID
+func createEventProperties(connectionID, invitationID string) interface{} { //nolint: unparam
+	return &didExchangeEvent{
+		connectionID: connectionID,
+		invitationID: invitationID,
+	}
 }
 
-// InvitationID returns DIDExchange invitationID.
-func (ex *didExchangeEvent) InvitationID() string {
-	return ex.invitationID
+func createErrorEventProperties(connectionID, invitationID string, err error) interface{} { //nolint: unparam
+	return &didExchangeEventError{
```

Comment:
call `createEventProperties` and then construct `didExchangeEventError{}` easy to maintain.

Reply:
[Done](https://github.com/hyperledger/aries-framework-go/compare/abea54bccf94c6f84b362bc1673c2639e2ecfa9c..05e7cc5e7e99a90abc09704d05394f0fb8ea61ba#diff-f761c91e753eba1c42a68076df06ce12R235).